src/main/python/docs/source/guide/python_end_to_end_tut.rst

```diff
@@ -0,0 +1,561 @@
+In our particular case we obtain the Matrix X1 and the Frame M1 from the operation. X1 holds all the encoded values and M1 holds a mapping between the encoded values
+and all the initial values. Columns that have not been specified in the .json file were not altered.
+
+.. code-block:: python
+
+    X1, M1 = X1.transform_encode(spec=jspec)
+
+We now can use the previously parsed dml file for splitting the dataset and unifying the inconsistent labels. It is noteworthy that the
+file is parsed such that we can directly call the function names from the Python API.
+
+.. code-block:: python
+
+    X = PREPROCESS_package.get_X(X1, 1, train_count)
+    Y = PREPROCESS_package.get_Y(X1, 1, train_count)
+
+    Xt = PREPROCESS_package.get_X(X1, train_count, train_count+test_count)
+    Yt = PREPROCESS_package.get_Y(X1, train_count, train_count+test_count)
+
+    Yt = PREPROCESS_package.replace_value(Yt, 3.0, 1.0)
+    Yt = PREPROCESS_package.replace_value(Yt, 4.0, 2.0)
+
+Step 4: Training and confusion matrix
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that we prepared the data we can use the multiLogReg function.
+These steps are identical to step 2 and 3 that have already been described in level 1 of this tutorial.
+
+.. code-block:: python
+
+    from systemds.operator.algorithm import multiLogReg
+    from systemds.operator.algorithm import confusionMatrix
+    from systemds.operator.algorithm import multiLogRegPredict
+    betas = multiLogReg(X, Y)
+    [_, y_pred, acc] = multiLogRegPredict(Xt, betas, Yt)
+    confusion_matrix_abs, _ = confusionMatrix(y_pred, Yt).compute()
+    print(confusion_matrix_abs)
+
+Full Script
+~~~~~~~~~~~
+
+The complete script now can be seen here:
+
+.. code-block:: python
+
+    import numpy as np
+    from systemds.context import SystemDSContext
+    from systemds.operator.algorithm import multiLogReg, multiLogRegPredict, confusionMatrix
+
+    train_count = 32561
+    test_count = 16281
+
+    dataset_path_train = "adult/train_data.csv"
+    dataset_path_test = "adult/test_data.csv"
+    dataset_jspec = "adult/jspec.json"
+    preprocess_src_path = "preprocess.dml"
+
+    sds = SystemDSContext()
+
+    SCHEMA = '"DOUBLE,STRING,DOUBLE,STRING,DOUBLE,STRING,STRING,STRING,STRING,STRING,DOUBLE,DOUBLE,DOUBLE,STRING,STRING"'
+
+    F1 = sds.read(dataset_path_train, schema=SCHEMA)
+    F2 = sds.read(dataset_path_test,  schema=SCHEMA)
+
+    jspec = sds.read(dataset_jspec, data_type="scalar", value_type="string")
+    PREPROCESS_package = sds.source(preprocess_src_path, "preprocess", print_imported_methods=True)
+
+    X1 = F1.rbind(F2)
+    X1, M1 = X1.transform_encode(spec=jspec)
+
+    X = PREPROCESS_package.get_X(X1, 1, train_count)
+    Y = PREPROCESS_package.get_Y(X1, 1, train_count)
+
+    Xt = PREPROCESS_package.get_X(X1, train_count, train_count+test_count)
+    Yt = PREPROCESS_package.get_Y(X1, train_count, train_count+test_count)
```

Comment:
It would be nicer if we use the split function to construct the train and test datasets,

Reply:
then we would not need to use source here in level 2, and make it cleaner i now understand why you commented on this in the emails